test/models/order_test.rb

```diff
@@ -0,0 +1,227 @@
+      order_one.valid?.must_equal false
+
+      order_one.card_cvv = 1234
+      order_one.valid?.must_equal false
+    end
+
+    it "must have a card cvv to be valid" do
+      order_one.valid?.must_equal true
+      order_one.card_cvv = nil
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :card_cvv
+
+      order_one.card_cvv = " "
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :card_cvv
+    end
+
+    it "must have a valid zip code to be true" do
+      order_one.valid?.must_equal true
+      order_one.zip_code = 9812500000
+      order_one.valid?.must_equal false
+
+      order_one.zip_code = 9812
+      order_one.valid?.must_equal false
+
+      order_one.zip_code = "98125-"
+      order_one.valid?.must_equal false
+    end
+
+    it "must have a zip code to be valid" do
+      order_one.valid?.must_equal true
+      order_one.zip_code = nil
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :zip_code
+
+      order_one.zip_code = " "
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :zip_code
+    end
+
+    it "must have a city to be valid" do
+      order_one.valid?.must_equal true
+      order_one.city = nil
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :city
+
+      order_one.city = " "
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :city
+    end
+
+    it "must have a state to be valid" do
+      order_one.valid?.must_equal true
+      order_one.state = nil
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :state
+
+      order_one.state = " "
+      order_one.valid?.must_equal false
+      order_one.errors.messages.must_include :state
+    end
+  end
+
+  describe "shipped" do
+    it "returns true if the order has a status of shipped" do
+      orders(:shipped_order).shipped.must_equal true
+    end
+    it "returns false if the order does not have a status of shipped" do
+      orders(:pending_order).shipped.must_equal false
+    end
+  end
+
+  describe "completed?" do
+    it "returns true if the order has a status of complete" do
+      orders(:order_one).completed?.must_equal true
+    end
+    it "returns false if the order does not have a status of complete" do
+      orders(:pending_order).shipped.must_equal false
+    end
+  end
+
+  describe "order_total" do
+    it "returns an accurate sum of the prices of the products connected to the order" do
+      order = orders(:pending_order)
+      product1 = products(:pointy_hat)
+      product2 = products(:sad_hat)
+
+      order.order_total.must_equal 0
+      order.products << product1
+      order.order_total.must_equal 9.99
+      order.products << product1
+      order.order_total.must_equal (9.99 * 2)
+      order.products << product2
+      order.order_total.must_equal (9.99 * 2) + 8.99
```

Comment:
What happens if the test data changes? Instead of using magic numbers/hardcoded numbers, it would be better if you use the data like `product1.price` instead of `9.99`